src/Microsoft.OData.Edm/Schema/EdmModelBase.cs

```diff
@@ -154,13 +161,31 @@ public IEnumerable<IEdmOperation> FindDeclaredOperations(string qualifiedName)
         /// <returns> A set of operations that share the binding type or empty enumerable if no such operation exists. </returns>
         public virtual IEnumerable<IEdmOperation> FindDeclaredBoundOperations(IEdmType bindingType)
         {
-            foreach (IEnumerable<IEdmOperation> operations in this.functionDictionary.Values.Distinct())
+            IList<IEdmOperation> bindableOperations;
+
+            if (!this.bindableOperationsCache.TryGetValue(bindingType.FullTypeName(), out bindableOperations))
             {
-                foreach (IEdmOperation operation in operations.Where(o => o.HasEquivalentBindingType(bindingType)))
+                HashSet<IList<IEdmOperation>> operationList = new HashSet<IList<IEdmOperation>>();
+                bindableOperations = new List<IEdmOperation>();
+
+                foreach (IList<IEdmOperation> operations in this.functionDictionary.Values)
                 {
-                    yield return operation;
+                    if (operationList.Add(operations))

```

Comment:
>if (operationList.Add(operations)) [](start = 20, length = 34)

maybe add a comment that this is in order to ensure the list of operations is unique (do we know, under what conditions, it will not be unique?)